internal/thelma/cli/commands/bees/apply_schedule/apply_schedule_command.go

```diff
@@ -0,0 +1,231 @@
+			if slack, err := app.Clients().Slack(); err != nil {
+				log.Debug().Msgf("Couldn't send slack message to the owner, errored building slack client: %v", err)
+			} else if slack == nil {
+				log.Debug().Msgf("Couldn't send slack message to the owner, no slack client")
+			} else if cmd.options.dryRun {
+				log.Debug().Msgf("Skipped sending slack message due to this being a dry run")
+			} else {
+				markdown := fmt.Sprintf("Hey there, your <https://broad.io/beehive/r/environment/%s|%s> BEE has a start/stop schedule with times too close together, so Thelma skipped applying the schedule just now. Thelma was using a range of %s, so try making the start and stop times at least that far apart.",
+					matchingBee.Name(), matchingBee.Name(), cmd.options.fromPast.String())
+				if err := slack.SendDirectMessage(matchingBee.Owner(), markdown); err != nil {
+					log.Debug().Msgf("Couldn't send a slack message to %s: %v", matchingBee.Owner(), err)
+					if err := slack.SendDevopsAlert("Conflicting Bee Schedule", fmt.Sprintf("BEE %s had conflicting start/stop times and an non-Slackable owner; try making the times at least %s apart", matchingBee.Name(), cmd.options.fromPast), false); err != nil {
+						log.Debug().Msgf("Couldn't send a devops slack alert: %v", err)
+					} else {
+						log.Debug().Msgf("Successfully send a devops slack alert")
+					}
+				} else {
+					log.Debug().Msgf("Successfully sent a slack message to %s", matchingBee.Owner())
+				}
+			}
+		} else if (!matchingBee.Offline() && wouldStop) || (matchingBee.Offline() && wouldStart) {
+			beesToFlip = append(beesToFlip, matchingBee)
+		}
+	}
+
+	if len(beesToFlip) == 0 {
+		log.Info().Msg("Found no matching BEEs with scheduling that needed applying")
+		return nil
+	}
+
+	state, err := app.State()
+	if err != nil {
+		return err
+	}
+
+	var successfullyFlippedEnvNames []string
+	var mutex sync.Mutex
+
+	var jobs []pool.Job
+	for _, unsafe := range beesToFlip {
+		env := unsafe
+		if cmd.options.dryRun {
+			log.Info().Msgf("Would've flipped %s target state to offline=%t but dry run was enabled", env.Name(), !env.Offline())
+		} else if err := state.Environments().SetOffline(env.Name(), !env.Offline()); err != nil {
+			log.Warn().Msgf("Failed to flip %s target state to offline=%t: %v", env.Name(), !env.Offline(), err)
+			continue
+		} else {
+			log.Info().Msgf("Flipped %s target state to offline=%t", env.Name(), !env.Offline())
+		}
+		jobs = append(jobs, pool.Job{
+			Name: env.Name(),
+			Run: func(_ pool.StatusReporter) error {
+				if cmd.options.dryRun {
+					log.Info().Msgf("Would've synced %s but dry run was enabled", env.Name())
+				} else {
+					log.Info().Msgf("Syncing %s", env.Name())
+					_, err := bees.SyncArgoAppsIn(env, func(options *argocd.SyncOptions) {
+						options.SkipLegacyConfigsRestart = true
+					})
+					if err != nil {
+						return err
+					}
+				}
+
+				mutex.Lock()
```

Comment:
```suggestion
				mutex.Lock()
				defer mutex.Unlock()
```
I don't think it matters here but in general I think it's considered preferable to handle mutexes like this

Reply:
Unlock will still be called before the function returns